aiohttp/streams.py

```diff
@@ -143,18 +142,22 @@ def __repr__(self) -> str:
             info.append("low=%d high=%d" % (self._low_water, self._high_water))
         if self._waiter:
             info.append("w=%r" % self._waiter)
-        if self._exception:
-            info.append("e=%r" % self._exception)
+        if self._waiter:
+            wait_exc = self._waiter.exception()
+            if wait_exc is not None:
+                info.append("e=%r" % wait_exc)
         return "<%s>" % " ".join(info)
 
     def get_read_buffer_limits(self) -> Tuple[int, int]:
         return (self._low_water, self._high_water)
 
     def exception(self) -> Optional[BaseException]:
-        return self._exception
+        if self._waiter is not None:
+            return self._waiter.exception()
```

Comment:
...and this, also probably have issues with `InvalidStateError`.

Reply:
Fixed